docs/extend/extend-cli/cli-developing-a-plugin.md

```diff
@@ -2,218 +2,339 @@
+7. Return to your terminal and run `cd files-util` to enter the project directory.
+8. Enter `git init` to set up a new Git repository.
+9. Enter `git add --all` to *stage* (track) all files in the current directory with Git.
 
-Completed source for this tutorial can be found on the `develop-a-plugin` branch of the zowe-cli-sample-plugin repository.
+10. Enter `git commit --message "Initial commit"` to save a snapshot of the staged files in your repository.
+11. Run `npm install` to install third-party dependencies defined in the `package.json` file of your Node.js project.
 
-### Cloning the sample plug-in source
- Clone the sample repo, delete the irrelevant source, and create a brand new plug-in. Follow these steps:
+    When successful, **[WHAT IS THE USER SUPPOSED TO SEE?]**.
 
-1. `cd` into your `zowe-tutorial` folder
-2. `git clone https://github.com/zowe/zowe-cli-sample-plugin files-util`
-3. `cd files-util`
-4. Delete the `.git` (hidden) folder.
-5. Delete all content within the `src/api`, `src/cli`, and `docs` folders.
-6. Delete all content within the `__tests__/__system__/api`, `__tests__/__system__/cli`, `__tests__/api`, and `__tests__/cli` folders
-7. `git init`
-8. `git add .`
-9. `git commit -m "initial"`
+**To create a unique plug-in:** Change the `files-util` directory to a name applicable for your project.
 
-### Changing package.json
-Use a unique `npm` name for your plugin. Change `package.json` name field as follows:
+## Updating `package.json`
+
+Change the name property in the `package.json` file to the plug-in name.
+
+Open the `package.json` file in a text editor and replace the name field with the following information:
 
 ```typescript
   "name": "@zowe/files-util",
 ```
 
-Issue the command `npm install` against the local repository.
+This tutorial uses `@zowe/files-util` as the tutorial plug-in name.
+
+**To create a unique plug-in:** Replace `@zowe/files-util` with a unique plug-in name. This allows you to publish the plug-in under that name to the `npm` registry in the future.
+
+## Adjusting Imperative CLI Framework configuration
+Define json configurations for the plug-in to Imperative.
+
+Change the `src/imperative.ts` file to contain the following configurations:
 
-### Adjusting Imperative CLI Framework configuration
-Change `imperative.ts` to contain the following:
 ```typescript
 import { IImperativeConfig } from "@zowe/imperative";
 
 const config: IImperativeConfig = {
     commandModuleGlobs: ["**/cli/*/*.definition!(.d).*s"],
     rootCommandDescription: "Files utility plugin for Zowe CLI",
-    envVariablePrefix: "FILES_UTIL_PLUGIN",
-    defaultHome: "~/.files_util_plugin",
     productDisplayName: "Files Util Plugin",
     name: "files-util"
 };
 
 export = config;
 ```
-Here we adjusted the description and other fields in the `imperative` JSON configuration to be relevant to this plug-in.
 
-### Adding third-party packages
+When successful, **[WHAT IS THE USER SUPPOSED TO SEE?]**.
```

Comment:
We either instruct them how to start testing their plugin changes so that they can see these updates "live" ...
OR
Remove this `when successful` section since the only thing we expect the user is to copy-paste the above code block and replace the existing `src/imperative.ts` file.